stdlib/public/SDK/Foundation/UUID.swift

```diff
@@ -83,8 +99,8 @@ public struct UUID : ReferenceConvertible, Hashable, Equatable, CustomStringConv
     
     fileprivate var reference: NSUUID {
         var bytes = uuid
-        return withUnsafePointer(to: &bytes) {
-            return NSUUID(uuidBytes: unsafeBitCast($0, to: UnsafePointer<UInt8>.self))
+        return withUnsafeMutablePointer(to: &bytes, as: UInt8.self, capacity: MemoryLayout.size(ofValue: bytes)) {
+            return NSUUID(uuidBytes: $0)
         }
```

Comment:
Nice, I didn't know about that form of `withUnsafePointer`. That might be a better recommendation for the compiler to emit a warning for when bitcasting from a homogeneous tuple to its element type. Mind filing a bug for that diagnostic improvement?